src/databricks/labs/ucx/progress/install.py

```diff
@@ -0,0 +1,51 @@
+import datetime as dt
+import logging
+from dataclasses import dataclass
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.labs.lsql.deployment import SchemaDeployer
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class WorkflowRun:
+    started_at: dt.datetime
+    """The timestamp of the workflow run start."""
+
+    finished_at: dt.datetime
+    """The timestamp of the workflow run end."""
+
+    workspace_id: int
+    """The workspace id in which the workflow ran."""
+
+    workflow_name: str
+    """The workflow name that ran."""
+
+    workflow_id: int
+    """"The workflow id of the workflow that ran."""
+
+    workflow_run_id: int
+    """The workflow run id."""
+
+    run_as: str
+    """The identity the workflow was run as`"""
+
+    status: str
+    """The workflow run final status"""
+
+
+class ArtifactInstallation:
+    """Install resources for UCX's artifacts."""
+
+    _SCHEMA = "artifacts"
+
+    def __init__(self, sql_backend: SqlBackend, ucx_catalog: str) -> None:
+        # `mod` is required parameter, but it's not used in this context.
+        self._schema_deployer = SchemaDeployer(sql_backend, self._SCHEMA, mod=None, catalog=ucx_catalog)
```

Comment:
Requires lsql release with: https://github.com/databrickslabs/lsql/pull/296

Reply:
released in v0.12.1